crates/node/p2p/src/net/driver.rs

```diff
@@ -110,6 +121,63 @@ impl Network {
                         };
                         self.gossip.dial(enr);
                     },
+
+                    _ = peer_score_inspector.tick(), if self.gossip.peer_monitoring.as_ref().is_some() => {
+                        // Inspect peer scores and ban peers that are below the threshold.
+                        let Some(ban_peers) = self.gossip.peer_monitoring.as_ref() else {
+                            continue;
+                        };
+
+                        // We iterate over all connected peers and check their scores.
+                        // We collect a list of peers to remove
+                        let peers_to_remove = self.gossip.swarm.connected_peers().filter_map(
+                            |peer_id| {
+                                 // If the score is not available, we use a default value of 0.
+                                 let score = self.gossip.swarm.behaviour().gossipsub.peer_score(peer_id).unwrap_or_default();
+
+                                 if score < ban_peers.ban_threshold {
+                                    return Some(*peer_id);
+                                 }
+
+                                 None
+                            }
+                        ).collect::<Vec<_>>();
+
+                        // We remove the addresses from the gossip layer.
+                        let addr_to_remove = peers_to_remove.into_iter().filter_map(|peer_to_remove| {
+                            // In that case, we ban the peer. This means...
+                            // 1. We remove the peer from the network gossip.
+                            // 2. We ban the peer from the discv5 service.
+                            if self.gossip.swarm.disconnect_peer_id(peer_to_remove).is_err() {
+                                warn!(peer = ?peer_to_remove, "Trying to disconnect a non-existing peer from the gossip driver.");
+                            }
+
+                            if let Some(addr) = self.gossip.peerstore.remove(&peer_to_remove){
+                                self.gossip.dialed_peers.remove(&addr);
+
+                                return Some(addr);
+                            }
+
+                            None
+                        }).collect::<HashSet<_>>();
+
+                        let ban_futures = handler.table_enrs().await.into_iter().filter_map(
```

Comment:
I think we'll want to spin this into a thread by sending a message through the `handler`. Otherwise this will block the main network driver here which process gossip messages.